src/overlays/actors/ovl_En_Guruguru/z_en_guruguru.c

```diff
@@ -21,32 +45,358 @@ const ActorInit En_Guruguru_InitVars = {
+    if (this->unk270 != 0) {
+        Player* player = PLAYER;
+
+        this->textIDIndex = 3;
+        if (player->transformation == PLAYER_FORM_DEKU) {
+            this->textIDIndex = 13;
+            if (gSaveContext.weekEventReg[79] & 4) {
+                this->textIDIndex = 14;
+            }
+        }
+
+        this->actor.textId = textIDs[this->textIDIndex];
+    }
+
+    yawTemp = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    yaw = ABS_ALT(yawTemp);
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_80BC701C(this, globalCtx);
+    } else if (yaw < 0x2891) {
+        func_800B8614(&this->actor, globalCtx, 60.0f);
+    }
+}
+
+void func_80BC701C(EnGuruguru* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->unk268 != 0) &&
+        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_DEKU))) {
+        this->headZRotTarget = 5000;
+    }
+
+    this->unk266 = 1;
+    this->unk272 = 1;
+    this->actionFunc = func_80BC7068;
+}
+
+void func_80BC7068(EnGuruguru* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/EnGuruguru_Destroy.asm")
+    if (this->unk268 != 0) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    } else if (this->unusedTimer == 0) {
+        this->unusedTimer = 6;
+        if (func_80152498(&globalCtx->msgCtx) != 5) {
+            if (this->unk266 == 0) {
+                if (this->headZRotTarget != 0) {
+                    this->headZRotTarget = 0;
+                } else {
+                    this->headZRotTarget = 5000;
+                }
+            }
+        } else {
+            if ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_DEKU)) {
+                this->headZRotTarget = 5000;
+            } else {
+                this->headZRotTarget = 0;
+            }
+        }
+    }
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && (func_80147624(globalCtx))) {
+        func_801477B4(globalCtx);
+        this->headZRotTarget = 0;
+        if ((this->textIDIndex == 13) || (this->textIDIndex == 14)) {
+            func_80151BB4(globalCtx, 0x13);
+            gSaveContext.weekEventReg[79] |= 4;
+            func_80BC6E10(this);
+            return;
+        }
+        if (this->actor.params == 0) {
+            if (this->actor.textId == 0x295F) {
+                gSaveContext.weekEventReg[88] |= 4;
+            }
+            if (this->actor.textId == 0x292A) {
+                gSaveContext.weekEventReg[38] |= 0x10;
+            }
+            func_80151BB4(globalCtx, 0x13);
+            func_80BC6E10(this);
+            return;
+        }
+        if (this->textIDIndex == 0xB) {
+            func_80BC73F4(this);
+            return;
+        }
+        if (this->textIDIndex == 0xC) {
+            gSaveContext.weekEventReg[38] |= 0x40;
```

Comment:
This whole function feels like a lot of chained if-else

Reply:
Agreed. Also consider using dec for `textIdIndex`?

Reply:
If I make these chained if-elses, my diff is missing a bunch of `lw ra`s that vanillas has, so I'm not sure if it is. I'll try a few things though